napari/layers/_shapes_layer/model.py

```diff
@@ -944,6 +952,28 @@ def _finish_drawing(self):
                 self.slice_data.remove(index)
         self._is_creating = False
         self.refresh()
+        self._update_thumbnail()
+
+    def _update_thumbnail(self):
+        """Update thumbnail with current points and colors.
+        """
+        offset = np.array([self.range[-2][0], self.range[-1][0]]) - 0.5
```

Comment:
Can you add comments here about what `self.range` holds? I'm confused by both the offset and the +1 in the lines below.

Reply:
have added comments - range holds the min / max values of the shape data. offset holds the min values, and the +1 is there for padding. all this is explained in the comments